src/github.com/stellar/horizon/resource/account.go

```diff
@@ -71,3 +72,16 @@ func (this *Account) Populate(
 	this.Links.Data.PopulateTemplated()
 	return
 }
+
+// GetData returns decoded value for a given key. If the key does
+// not exist, empty slice will be returned.
+func (this *Account) GetData(key string) []byte {
+	data, exists := this.Data[key]
+	if !exists {
+		return nil
+	}
+
+	// We assume that horizon always returns valid base64 encoded string
+	decoded, _ := base64.StdEncoding.DecodeString(data)
```

Comment:
> You mean who? A developer/hacker? If the value is not valid base64 encoded string then it's a bug, right?

I mean that this data is being loaded from a database that isn't under control by horizon.  This data is written by stellar-core.  stellar-core is an external system to horizon and we should be validating the data stellar-core produces where we can.  Every place we don't represents an unconditional trust in stellar-core.

> you can start thinking that GetBytes isn't that helpful since encoding/base64 is a standard library

GetData's primary use should be to prevent misinterpretation... the consumer need not know that the data is encoded as base64 to consume it.  It simply calls the method and gets the raw data back.  

> Whereas, GetData(key string) []byte allows you to do certain things like passing the result to the function directly what's not possible for multiple return functions (ex. fmt.Println(string(account.GetBytes("key")))).

The way to turn an error producing func into a non-error producing func is to produce a `Must*` wrapping.  Otherwise, you pass along the failure and end up confusing the poor soul who has to debug the system.  Let's take your example and pretend that the "key" value has been truncated in such that decoding it fails.  The example statement will produce the output:

```

```

But if you were to look into the stellar-core database, you'll clearly see that the column is populated with some data!  You'd then have to try decoding the value yourself to understand that an error was occurring.  What happens if the line was written `fmt.Println(string(account.MustGetBytes("key"))))`?  You'd get a stack trace whose top line says "illegal base64 data at input byte N`.

The real scope of the problem comes into focus when you aren't just printing debug information... the issue compounds when you're doing work with the corrupted return value. You are forced to work your way down from the unexpected outcome (such as an empty string in the http response), stack frame by stack frame until you see that there was a call to `GetBytes` that was swallowing the error.  That isn't fun debugging.

> My main assuption here is that: we should trust horizon to have valid, consistent data.

A mentioned... this isn't horizon data.  It's stellar-core data.

> Otherwise, we would have to check every single value returned by it before using it in other apps code including:

We validate data in two situations: where we must, and where we can when it's prudent.  We _must_ validate data that isn't owned by horizon.  I'd be happy to accept PRs adding validation where we aren't already. Even for data owned by horizon, we should validate the data when we can.  Database administration happens, and sometimes that means db corruption.  In that situation and when its prudent, we should fail rather than ignore an error.

If we could prudently check your whole list, I would argue we should. Indeed many of the examples you provide _are_ validated by horizon when the data is coming from an outside source. This is financial software bartek!  We should have layers of defense in our code and architectures. We should try to consider and compensate for every possible scenario... if stellar becomes widely adopted it will get all sorts of attacks thrown its way.